### PR TITLE
Add option to use project templates to initialize a repository

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
Add option to use project templates to initialize a repository

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/getting-started/examples.md [hunk 1]
```diff
@@ -41,6 +41,14 @@ repository of ZenML.
 
 [Click here to go to ZenML Projects](https://github.com/zenml-io/zenml-projects)
 
+## ZenML Project Templates: Generate your own ZenML projects
+
+Use one of our ZenML Project Templates to generate ready-to-use scaffolds for
```
Comment (strickvl) [inline docs/book/getting-started/examples.md:46]:
```suggestion
Use one of our ZenML Project Templates to generate ready-to-use scaffolding for
```

File: docs/book/getting-started/examples.md [hunk 2]
```diff
@@ -41,6 +41,14 @@ repository of ZenML.
...
+your own projects, with everything you need to get started: ZenML steps,
+pipelines, stack configurations and other useful resources.
+
+[Click here to go to ZenML Projects](https://github.com/zenml-io/zenml-project-templates)
```
Comment (strickvl) [inline docs/book/getting-started/examples.md:50]:
```suggestion
[Click here to go to ZenML Templates](https://github.com/zenml-io/zenml-project-templates)
```

File: pyproject.toml [hunk 3]
```diff
@@ -79,8 +79,16 @@ fastapi-utils = { version = "~0.2.1", optional = true}
 
 # optional dependencies for stack recipes
 
+# Optional dependencies for project templates
+copier = ">=7.0.0"
+jinja2-time = "^0.2.0"
+black = "^22.3.0"
+ruff = "^0.0.238"
+autoflake = "^2.0.1"
```
Comment (strickvl) [inline pyproject.toml:87]: I think we don't need autoflake any more. We just removed it for ourselves [in this PR](https://github.com/zenml-io/zenml/pull/1279).

File: pyproject.toml [hunk 4]
```diff
@@ -79,8 +79,16 @@ fastapi-utils = { version = "~0.2.1", optional = true}
 
 # optional dependencies for stack recipes
 
+# Optional dependencies for project templates
+copier = ">=7.0.0"
```
Comment (schustmi) [inline pyproject.toml:83]: I think these are also all non-optional right now? See the `optional=true` used above

**Review by schustmi — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (stefannica) [reply pyproject.toml]: Beautiful ! I updated my templates scripts to reflect this. One package less in my requirements.

File: pyproject.toml (hunk 4 above)
Comment (stefannica) [reply pyproject.toml]: Yes, correct. I made them all optional.

**Review by schustmi — APPROVED** [review]

**Review by strickvl — APPROVED** [review]

**Pull request merged**